pkg/kubefed2/unjoin.go

```diff
@@ -144,7 +144,12 @@ func (j *unjoinFederation) Run(cmdOut io.Writer, config util.FedConfig) error {
 	clusterConfig, err := config.ClusterConfig(j.clusterContext, j.Kubeconfig)
 	if err != nil {
 		glog.V(2).Infof("Failed to get unjoining cluster config: %v", err)
-		return err
+
+		if !j.forceDeletion {
+			return err
+		}
+		// if member cluster is not available with forceDeletion,
+		// deleting federation information in the host cluster for the unjoining cluster
```

Comment:
-> If configuration for the member cluster cannot be successfully loaded, forceDeletion indicates that resources associated with the member cluster should still be removed from the host cluster.